### PR TITLE
Add missing UID in SubjectAccessReviewSpec

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
WebhookAuthorizer's Authorize should send *all* the information
present in the user.Info data structure. We are not sending the
UID currently.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
The SubjectAccessReview API in the authorization.k8s.io API group now allows providing the user uid.
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func (w *WebhookAuthorizer) Authorize(attr authorizer.Attributes) (authorized bo
 	if user := attr.GetUser(); user != nil {
 		r.Spec = authorization.SubjectAccessReviewSpec{
 			User:   user.GetName(),
+			UID:	user.GetUID(),
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go]: a corresponding change in sarApprover.authorize and the CSR API propagating the uid would make sense to me

Comment (ericchiang) [inline staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go]: @dims want to take that in this PR? If not i can send one.

Comment (dims) [inline staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go]: Fixed a few more spots. Will wait for the CI to run to see if the changes hold up

**Comment by liggitt** [comment]
need to run `hack/update-all.sh` to regenerate everything associated with an API change

**Comment by dims** [comment]
On it. @liggitt

File: staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go (hunk 1 above)
Comment (dims) [inline staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go]: @ericchiang i found the "sarApprover.authorize" but not sure if i have covered all the cases, please see latest patch.

**Comment by liggitt** [comment]
cc @cjcullen for a new field sent to the authz webhook

**Comment by liggitt** [comment]
/lgtm
Needs approval

**Comment by smarterclayton** [comment]
/approve

This is relevant info and is important.  It was not omitted intentionally.

@dims are there other endpoints that need this like external web hooks?

**Comment by dims** [comment]
@smarterclayton : thanks a ton. I will review more to see where else we are missing UID. i was focused on SubjectAccessReview, but will widen the net.

**Pull request merged**